portal/src/graphql/portal/AdminAPIConfigurationScreen.tsx

```diff
@@ -167,8 +189,51 @@ const AdminAPIConfigurationScreenContent: React.VFC<AdminAPIConfigurationScreenC
       [navigate, items]
     );
 
+    const generateShortLivedAdminAPITokenHandle = useCallback(async () => {
+      if (secretToken == null) {
+        // generateShortLivedAdminAPITokenHandle should be called only
+        // when there is a secret token
+        console.error("secret token should not be null");
+        return;
+      }
+      const token = await generateShortLivedAdminAPIToken(secretToken);
+      setShortLivedAdminAPIToken(token);
+    }, [generateShortLivedAdminAPIToken, secretToken]);
+
+    const onClickGenerateShortLivedAdminAPIToken = useCallback(() => {
+      setShortLivedAdminAPIToken(null);
+
+      const reauthentication = () => {
+        const state: LocationState = {
+          keyID: "",
+          shouldRefreshSecretToken: true,
+          shouldGenerateShortLivedAdminAPIToken: true,
+        };
+
+        startReauthentication(navigate, state).catch((e) => {
+          console.error(e);
+        });
+      };
+      // There are two possible cases:
+      // 1. secretToken is null
+      // 2. secretToken is not null but failed to generate short-lived admin API token
+      // in both cases, reauthentication is needed
+      if (secretToken != null) {
+        generateShortLivedAdminAPITokenHandle().catch(() => {
+          reauthentication();
+        });
```

Comment:
Can we check the actual error thrown, and only call `reauthentication` if the error is related?
I think there might be other errors which we want to log an error and probably display error ui  to user. e.g. Network error or server error.